cmake/process_options.cmake

```diff
@@ -117,70 +117,30 @@ endif()
 
 if(ENABLE_KOKKOS)
     message(STATUS "ENABLE_KOKKOS is ON.")
-    find_library( KOKKOS_CORE_STATIC
-    NAMES   libkokkoscore.a
-    HINTS   /usr/lib
-            /usr/local/lib
-            /opt
-            lib
-            lib64
-            ${CMAKE_SOURCE_DIR}/lib
-            ${CMAKE_SOURCE_DIR}/lib64
-            ENV KOKKOS_LIB
-            ENV LD_LIBRARY_PATH
-    )
-    find_library( KOKKOS_KERNELS_STATIC
-    NAMES   libkokkoskernels.a
-    HINTS   /usr/lib
-            /usr/local/lib
-            /opt
-            lib
-            lib64
-            ${CMAKE_SOURCE_DIR}/lib
-            ${CMAKE_SOURCE_DIR}/lib64
-            ENV KOKKOS_LIB
-            ENV LD_LIBRARY_PATH
-    )
-    find_file( KOKKOS_CORE_INC
-    NAMES   Kokkos_Core.hpp
-    HINTS   /usr/include
-            /usr/local/include
+
+    find_package(Kokkos
```

Comment:
Can we guarantee the existence of the compiled library with this? Not sure if this is a better approach.

Reply:
I believe that we can. Even in Windows CMake was able to detect the libraries.

Reply:
Could you please elaborate on the reason why this is not a good approach?

Reply:
I mean, the reason why you are not sure if this is a good approach.

Reply:
Do we not need a separate `FindKokkos.cmake` file for this functionality to work, wherein we specify how it finds the required libraries and headers?

Reply:
I don't think so. According to Kokkos, they provide everything that is needed.
This is their suggested method of incorporating the already installed Kokkos and Kokkos Kernels packages into a third-party building process.

Reply:
Are there examples of where this happens in other repos? Just curious how this works, as I expected the [docs](https://cmake.org/cmake/help/latest/command/find_package.html) seem to mention the path it investigates for the .cmake file.

Though, if it just works, that is fine by me.

Reply:
Yes. But we are not supposed to make a file for that. In the same way, we are not supposed to make a file for finding BLAS, or OpenMP. 
Because the installation is not in a usual place, in the code, I hint to CMake about the Kokkos installation directory and CMake finds the files it needs there.

Reply:
The reason we don't for BLAS or OpenMP is because https://github.com/Kitware/CMake/blob/8d8957e56d8a58fd3966618eff742def509cfd39/Modules/FindBLAS.cmake and https://github.com/Kitware/CMake/blob/ddfccc36de642c96990d33a48dade0b00609169c/Modules/FindOpenMP.cmake exist natively to CMake.

But if it works like this, fine by me.

Reply:
Makes sense. Regarding Kokkos, by providing a hint CMake is able to find it.